tests/unit/tests_standard_lib/test_cim_tests/test_field_test_helper.py

```diff
@@ -0,0 +1,311 @@
+                "SEARCH_UTIL_RETURN_VALUE_1",
+                "SEARCH_UTIL_RETURN_VALUE_2",
+            ]
+        )
+        record_property.assert_called_once_with(
+            "search",
+            "| search (index=* OR index=_internal) | search tag=tag_splunkd_fiction_one",
+        )
+
+
+def test_gen_condition(mocked_field_test_helper):
+    mocked_field_test_helper.fields = [
+        field(condition="value > threshold_1"),
+        field(condition=""),
+        field(condition="value < threshold_2"),
+        field(condition="value = threshold_3"),
+    ]
+    assert (
+        mocked_field_test_helper._gen_condition()
+        == "value > threshold_1 OR value < threshold_2 OR value = threshold_3"
+    )
+
+
+@pytest.mark.parametrize(
+    "results, fields, expected_output",
+    [
+        (
+            [
+                {
+                    "sourcetype": "splunkd",
+                    "source": "/opt/splunk/var/log/splunk/health.log",
+                    "event_count": "45",
+                },
+            ],
+            None,
+            [
+                {
+                    "sourcetype": "splunkd",
+                    "source": "/opt/splunk/var/log/splunk/health.log",
+                    "event_count": 45,
+                },
+            ],
+        ),
+        (
+            [
+                {
+                    "sourcetype": "splunkd",
+                    "source": "/opt/splunk/var/log/splunk/license_usage.log",
+                    "event_count": "12",
+                    "field_1_count": "8",
+                    "field_1_valid_count": "5",
+                    "field_2_count": "9",
+                    "field_2_valid_count": "7",
+                    "field_2_invalid_values": "2",
+                },
+            ],
+            [
+                field(name="field_1", gen_validity_query=lambda: True),
+                field(name="field_2", gen_validity_query=lambda: True),
+            ],
+            [
+                {
+                    "sourcetype": "splunkd",
+                    "source": "/opt/splunk/var/log/splunk/license_usage.log",
+                    "event_count": 12,
+                    "field_count": 8,
+                    "valid_field_count": 5,
+                    "invalid_values": "-",
+                    "field_name": "field_1",
+                },
+                {
+                    "sourcetype": "splunkd",
+                    "source": "/opt/splunk/var/log/splunk/license_usage.log",
+                    "event_count": 12,
+                    "field_count": 9,
+                    "valid_field_count": 7,
+                    "invalid_values": "2",
+                    "field_name": "field_2",
+                },
+            ],
+        ),
+    ],
+)
+def test_parse_result(
+    field_test_adapter_mock, mocked_field_test_helper, results, fields, expected_output
+):
+    mocked_field_test_helper.fields = fields or []
+    if fields:
```

Comment:
Probably that can be avoided by defining global vars:
```
field_1 = field(name="field_1", gen_validity_query=lambda: True)
field_2 = field(name="field_2", gen_validity_query=lambda: True)
```
And using it in parametrisation directly